app/controllers/lesson_completions_controller.rb

```diff
@@ -1,44 +1,34 @@
 class LessonCompletionsController < ApplicationController
   
   before_filter :authenticate_request
-  
+  before_action :set_lesson, only: [:create, :destroy]
+  before_action :validate_lesson, only: [:create, :destroy]
+  before_action :set_course, only: [:create, :destroy]
+  before_action :set_next_lesson, only: [:create, :destroy]
+
   def create
     # Validate that the id does in fact correspond to a real lesson
     # Note that the authentication before_filter makes sure we've got a current_user
-    @lesson = Lesson.where(:id => params[:lesson_id]).first
 
-    if @lesson.nil?
-      render :nothing => true, :status => 400 # bad request
-    else   
-      @course = @lesson.course
-      @next_lesson = @lesson.next_lesson
-      lc = LessonCompletion.new(:student_id => current_user.id, :lesson_id => @lesson.id)
-      if lc.save
-        render "create", :formats => [:js]
-      else
-        render :nothing => true, :status => 400 # bad request
-      end
+    lc = LessonCompletion.new(:student_id => current_user.id, :lesson_id => @lesson.id)
```

Comment:
`lc` for me? is a magic variable, please define a better name for this variable?